modules/puppet_proxy/puppet_cache.rb

```diff
@@ -1,16 +1,17 @@
 require 'proxy/memory_store'
 require 'thread'
+require 'puppet_proxy/dependency_injection/container'
 
 module Proxy::Puppet
   class PuppetCache
     include Proxy::Log
+    extend Proxy::Puppet::DependencyInjection::Injectors
 
-    attr_reader :class_scanner
+    inject_attr :puppet_class_scanner_impl, :puppet_class_scanner
 
-    def initialize(class_scanner, classes_store, timestamps_store)
+    def initialize(classes_store = MemoryStore.new, timestamps_store = MemoryStore.new)
```

Comment:
This needs to be an explicit call to Proxy::MemoryStore, or you get "unitialized constant"